test/AutoDiff/differentiable_func_type_type_checking.swift

```diff
@@ -21,6 +21,16 @@ let _: @differentiable(linear) (Float) -> Float
 // Function conversion
 //===----------------------------------------------------------------------===//
 
+func nonescapingArgument(f: @differentiable (Float, Float) -> Float) -> Float {
+  return gradient(at: 1) { x in f(x, x) }
+}
+
+func nonescapingArgumentError(
+  f: @differentiable (Float, Float) -> Float
+) -> @differentiable (Float) -> Float{
+  return { x in f(x, x) }
```

Comment:
Shouldn't this case have an `expected-error`?

Reply:
I thought you added an error for this earlier. But now it means that `f` is not correctly diagnosed even with `followUses`.

Reply:
The issue is that this check happens in SIL. This file is the wrong place to put the test.